CONTRIBUTING.md

```diff
@@ -1,91 +1,379 @@
 
-6. If there is conflict, resolve the conflict, and then 
+3. If there is conflict, resolve the conflict, and then 
 
   ```bash
    git rebase --continue  
   ```
 
   After the code is successfully rebased, a window will pop up to edit the commit log, edit it then save and exit.
-7. After rebase, now you have a clean log history. push to your remote working branch
+4. After rebase, now you have a clean log history. push to your remote working branch
 
   ```
   git push origin branch_issueId
   ```
 
-  If commits have already been pushed to <git-user-id>/gearpump fork on github, you will have to "git push --force" to overwrite them with squashed commits.
+  If commits have already been pushed to your forked repository on GitHub, you will have to "git push --force" to overwrite them with squashed commits.
 
   ```
   git push origin -f branch_issueId
   ```
 
-8. Ensure all the unit tests are passed by running command "sbt test".
-6. Open a PR, which is a one-click thing in github.com; it knows you likely are opening a PR against upstream master. [Guide](https://help.github.com/articles/creating-a-pull-request) is here.
-7. [Merge PR](https://help.github.com/articles/merging-a-pull-request), [delete branch](https://help.github.com/articles/deleting-unused-branches).
+5. Ensure all the unit tests and integration tests are passed, check [Test](#test) for details.
+6. Open a Pull Request, which is a one-click thing in github.com; it knows you likely are opening a PR against upstream master. [Guide](https://help.github.com/articles/creating-a-pull-request) is here.
+
+<a name="code-review"></a>
+# Code Review
+Committer will review your code periodically. When there is any comment/feedback from committer(s), it's contributor's duty to update the pull request correspondingly. 
+
+When the merge is done by committer, you can optionally [delete your PR branch](https://help.github.com/articles/deleting-unused-branches).
+
+
+<a name="build-and-test"></a>
+
+# Build and test
+Though without a development environment setup, you can still contribute to Gearpump by reporting ideas, documentations, bugs and feature requests.
+It is highly recommended to set up a development environment to make any code contribution.
+
+<a name="local-copy"></a>
+## Clone Gearpump repository and make a local copy
+If you just want to study Gearpump source code, it is optional to perform following steps. 
+But, if you plan to contribute to Gearpump's code base, it is necessary to perform following steps:
+
+1. [Fork](https://help.github.com/articles/fork-a-repo) in GitHub from https://github.com/apache/incubator-gearpump to create a /incubator-gearpump repo. 
+After fork, you will have a new repo at https://github.com/your_username/incubator-gearpump.
```

Comment:
a bit wordy, How about "[Fork](https://help.github.com/articles/fork-a-repo) `https://github.com/apache/incubator-gearpump` to `https://github.com/your_username/incubator-gearpump`"